horovod/spark/runner.py

```diff
@@ -232,31 +283,130 @@ def run(fn, args=(), kwargs={}, num_proc=None, start_timeout=None,
+    if spark_context is None:
+        raise Exception('Could not find an active SparkContext, are you '
+                        'running in a PySpark session?')
+
+    if start_timeout is None:
+        # Lookup default timeout from the environment variable.
+        start_timeout = int(os.getenv('HOROVOD_SPARK_START_TIMEOUT', '600'))
+
+    # nics needs to be a set
+    if nics and not isinstance(nics, set):
+        nics = set(nics)
+
+    if num_proc is None:
+        # TODO: try spark.dynamicAllocation.initialExecutors
+        num_proc = spark_context.defaultParallelism
+        if verbose >= 1:
+            logging.info('Running %d processes (inferred from spark.default.parallelism)...', num_proc)
+    else:
+        if verbose >= 1:
+            logging.info('Running %d processes...', num_proc)
+
+    if min_np is None:
+        # TODO: try spark.dynamicAllocation.minExecutors
+        min_np = num_proc
+    if max_np is None:
+        # TODO: try spark.dynamicAllocation.maxExecutors
+        max_np = num_proc
+
+    # start Spark driver service and launch settings.num_proc Spark tasks
+    key = secret.make_secret_key()
+    spark_job_group = 'horovod.spark.run.%d' % job_id.next_job_id()
+    driver = driver_service.SparkDriverService(num_proc, fn, args, kwargs, key, nics)
+
+    discovery = host_discovery.SparkDriverHostDiscovery(driver)
+
+    tmout = timeout.Timeout(start_timeout,
+                            message='Timed out waiting for {activity}. Please check that you have '
+                                    'enough resources to run all Horovod processes. Each Horovod '
+                                    'process runs in a Spark task. You may need to increase the '
+                                    'start_timeout parameter to a larger value if your Spark resources '
+                                    'are allocated on-demand.')
+    settings = hvd_elastic_settings.ElasticSettings(num_proc=num_proc,
+                                                    min_np=min_np,
+                                                    max_np=max_np,
+                                                    discovery=discovery,
+                                                    verbose=verbose,
+                                                    key=key,
+                                                    start_timeout=tmout,
+                                                    elastic_timeout=elastic_timeout,
+                                                    nics=nics,
+                                                    run_func_mode=True)
+
+    result_queue = queue.Queue(1)
+
+    # launch settings.num_proc / settings.max_np Spark tasks
+    spark_thread = _make_spark_thread(spark_context, spark_job_group, driver,
+                                      result_queue, settings, use_gloo=True, is_elastic=True)
+    try:
+        # TODO: why are we doing this in static and is it needed for elastic?
```

Comment:
@alsrgv might be able to comment, as the original author of this code.

Reply:
I have removed those comments but would still like to know if this is something I need to do in elastic horovod. @alsrgv